app/routes/_protected.verify-email.($code).tsx

```diff
@@ -0,0 +1,28 @@
+import { useNavigate } from '@remix-run/react';
+import { VerifyEmailForm } from '~/features/signup/verify-email-form';
+import { shouldVerifyEmail as shouldUserVerifyEmail } from '~/features/user/user';
+import { useUserStore } from '~/features/user/user-provider';
+import { useEffectNoStrictMode } from '~/lib/use-effect-no-strict-mode';
+
+export default function VerifyEmail() {
+  const user = useUserStore((state) => state.user);
+  const navigate = useNavigate();
+  const shouldVerifyEmail = shouldUserVerifyEmail(user);
+
+  useEffectNoStrictMode(() => {
+    if (!shouldVerifyEmail) {
+      console.debug('Redirecting from verify email to /');
+      navigate('/');
+    }
+  }, [shouldVerifyEmail, navigate]);
+
+  if (!shouldVerifyEmail) {
+    return null;
+  }
+
+  return (
+    <div className="flex h-screen w-full items-center justify-center px-4">
+      <VerifyEmailForm user={user} />
```

Comment:
Should we get the verification  `code` param in this page and then pass it to VerifyEmailForm instead of getting the query param in the from component? That makes more sense to me because the name of this file is defining `code` as a param.

Reply:
sure. it does feel a bit cleaner

Reply:
changed